setup/mail-users.sh

```diff
@@ -20,10 +20,12 @@ db_path=$STORAGE_ROOT/mail/users.sqlite
 # Create an empty database if it doesn't yet exist.
 if [ ! -f "$db_path" ]; then
 	echo "Creating new user database: $db_path";
-	echo "CREATE TABLE users (id INTEGER PRIMARY KEY AUTOINCREMENT, email TEXT NOT NULL UNIQUE, password TEXT NOT NULL, extra, privileges TEXT NOT NULL DEFAULT '');" | sqlite3 "$db_path";
+	echo "CREATE TABLE users (id INTEGER PRIMARY KEY AUTOINCREMENT, email TEXT NOT NULL UNIQUE, password TEXT NOT NULL, extra, privileges TEXT NOT NULL DEFAULT '', quota TEXT NOT NULL DEFAULT '0');" | sqlite3 $db_path;
```

Comment:
If I understand correctly, '0' means unlimited and non-zero represents the quota. But there's no symbolic representation for "system default' from what I can see. So for example, if I made the system default 1GB, then the users would all get their DB value set to 1GB. If I later were to increase the system default to 10GB, the existing users would stay at 1GB and not inherit the new default.

Correct me if I've overlooked something and that's not how it works, but assuming that description is correct, then I'm wondering if the users that are getting the "system default quota" should have a symbolic representation of this in the DB, so if the default were to be changed, these users inherit the new value automatically.

Reply:
Quota is not applied, by default.   0 represents "no quota or 'unlimited' quota"     (obviously the limit is the size of the hard-drive for miab server.

Reply:
here is an example of setting the quota:

![miab_quota3](https://github.com/mail-in-a-box/mailinabox/assets/20545184/15058bb2-dd94-4b37-8867-36c78dcc2f69)

Then enter the number and and click the "set quota" button

![miab_quota2](https://github.com/mail-in-a-box/mailinabox/assets/20545184/7021e1c3-8a44-473e-bdc1-b2825842bdab)

Reply:
I think my confusion stems from the fact that 0 means unlimited, but I also see a `system default-quota` value that can be set, and I'm not sure what role that default value plays in this. How would I set a user to the "default-quota" value?

Reply:
@dms00 -  Will not having some of these answers prevent this from being merged?  Again, code is based on original code from PR -  https://github.com/mail-in-a-box/mailinabox/pull/1568 created by  @jrsupplee.    Of which we don't have all the answers as to why things were done the original way.   Perhaps this was meant for something that wasn't implemented.  I do appreciate you reviewing this though.  If its required for the PR to be completed we can try to figure out what John was doing.

Reply:
I've tried to use it but when I set quota and try to send a mail that has attachments to exceed the quota it errors but mail still sent but it didn't saved as sent folder.. How can I prevent that?

Reply:
So  you installed the fork, you set a quota.   Then you tried to send mail out of that mailbox in order to exceed the quota.   It sends the mail, but doesnt save the mail in the sent box.   

Am I understanding the above correctly?

You wouldn't prevent that, that is the expected result.   Once a user hits a quota that is set, they need to either remove mail from their mailbox, sent folder, trash, etc  or things become a bit unstable (by design) you wont be allowed to future things to disk as you exceed the quota.   

How can you prevent it,  give yourself more quota or delete mail.  

Maybe I dont understand correctly though....

Reply:
You understand correctly but I want to prevent that email sent. If quota is full user shouldn't be able to use it. And what can be do about 1 quota per domain? like example.com has 10GB total quota and every user in it like hi@example.com go@example.com has same quota based on domain but it will not affect to example.net?

Reply:
I understand the confusion around the default quota bits.  I have to look into this more.  As I understand it, default quota could either be the default quota for new users or the default quota for all users that don't have a quota set.  But since "not having a quota set" means "having your quota set to '0' because it's unlimited" then the question becomes: if, in fact, "default quota" is the quota for all users who don't have a quota set, then is it actually just ignored?  And, alternatively, if "default quota" is just for the new users, then is it actually more confusing than useful?

I agree, removing default quota seems to make sense.  I can look into this.  Likely just a UI adjustment.

Reply:
So there's a bit of a question here -- if we remove the default value of '0' from the table, it will need to be replaced with something otherwise all users will _require_ a quota.  Replacing it with empty string or Null makes the most sense, but i'm not sure how this will ripple out, so this is TBD?

Alternatively, if we remove the "system default" API endpoints, CLI commands, etc, this would reduce confusion while allowing users to continue to default to unlimited which I think makes the most sense.   

That said, ideally I could make the system default "unlimited" and then have a per-user quota and where users aren't specifically given a quota then when the system default changes the user's quota would also change for any user where the quota isn't set.  That said, I'm not at all sure how to do this with dovecot.  I can likely do this with the python / shell commands without too much trouble, but how dovecot will handle it is something I need to understand a bit better...

Reply:
fixed here: https://github.com/mail-in-a-box/mailinabox/pull/2387/commits/7c7b744605ed3a78c1c3f956601033bb547948d3 -- note that for this we just remove the ability to configure the system-default because it's confusing and default everyone to unlimited because it's simple.